src/Data/Container/Indexed/Relation/Binary/Pointwise.agda

```diff
@@ -0,0 +1,53 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Pointwise equality for indexed containers
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Container.Indexed.Relation.Binary.Pointwise where
+
+open import Data.Product using (_,_; Σ-syntax)
+open import Function
+open import Level using (Level; _⊔_)
+open import Relation.Binary using (REL; _⇒_)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl)
+
+open import Data.Container.Indexed.Core using (Container; Subtrees; ⟦_⟧)
+
+private variable
+  ℓᵉ ℓᵉ′ ℓᵖ ℓˢ ℓˣ ℓʸ : Level
+  I O : Set _
+
+-- Equality, parametrised on an underlying relation.
+
+module _ (C : Container I O ℓˢ ℓᵖ)
+         {X : I → Set ℓˣ} {Y : I → Set ℓʸ} (R : (i : I) → REL (X i) (Y i) ℓᵉ)
```

Comment:
Modulo an implict `i` then the type of R is:

https://github.com/agda/agda-stdlib/blob/358ad8bf673f0b292f5e5e86b50f0f6ec1d245cd/src/Relation/Binary/Indexed/Homogeneous/Core.agda#L32

Reply:
I tried your suggestion, but passing an `IREL` to `Pointwise` then needs an eta-expansion, like:
```agda
map : (R⇒R′ : ∀{i : I} → R {i} ⇒ R′ {i}) {o : O} → Pointwise C (λ{i} → R {i}) o ⇒ Pointwise C (λ{i} → R′ {i}) o
````
Maybe `IREL`s would be better off with a non-hidden argument?

Reply:
Hmm interesting. I've used `IRel` (rather than `IREL`) a reasonable amount, and Agda's always been able to infer the hidden arguments fine. Hmm I'll open an issue to address it.

Reply:
Of course, I might prefer the eta-expansion in the examples above to look like `λ {i} → R′ {i = i}` rather than `λ{i} → R′ {i}`, but is there a technical reason why Agda doesn't/can't eta-expand implicit functions? 

I've been meaning to raise an issue for Agda itself for a while, and it came up in the context of this discussion when we were reviewing outstanding issues for the `std-lib` v2.0 milestone, so I'll ask it here as a shortcut...